dagger-core/src/main/java/io/odpf/dagger/core/processors/external/http/HttpSourceConfig.java

```diff
@@ -90,6 +93,16 @@ public String getEndpoint() {
         return endpoint;
     }
 
+    /**
+     * Gets endpoint variables.
+     *
+     * @return the endpointVariables
+     */
+    public String getEndpointVariables() {
```

Comment:
we should use this for Get Request as well, but we will need to make it backwards compatible

Reply:
updated

Reply:
Also, I have a concern that this might break the existing running daggers when deployed. 

`Reason`:  As of now, the config `PROCESSOR_POSTPROCESSOR_CONFIG` looks something like below ( example)

```
{
  "external_source": {
	"http": [
    {
      "endpoint": "http://localhost:8080/match",
      "verb": "post",
      "request_pattern": "{\"locations\":%s,\"hints\”:{\”type\”:\”car\”,\”isold\”:false,\”isnew\”:true,\”percentage\”:50}}",
      "request_variables": “my_location”,
      "stream_timeout": "100000",
      "connect_timeout": "10000",
      "fail_on_errors": "false",
      "capacity": "30",
      "headers": {
        "content-type": "application/json"
      },
      "output_mapping": {
        “response_data”: {
          "path": "$"
        },
        “my_stage”: {
          "path": “$.stage”
        }
      }
    }
  ]
  }
}
```

This JSON config value is then deserialized into `PostProcessorConfig`  [here](https://github.com/odpf/dagger/blob/e46f634ca87de896b701ecd0acf1832b4b73ebdd/dagger-core/src/main/java/io/odpf/dagger/core/processors/PostProcessorConfig.java#L48). AFAIK, for existing daggers `endpointVariables` will then be deserialized into `null`. This may cause a `NullPointerException` when you are checking like this :

```
   if (httpSourceConfig.getEndpointVariables().length() > 0) {
            endpoint = String.format(httpSourceConfig.getEndpoint(), httpSourceConfig.getEndpointVariables());
        }
```

Reply:
> Any reason why endpointVariables is a String and not an Object[] like requestVariables ?

Can you point me where requestVariables is an Object[]? Since based on [this](https://github.com/odpf/dagger/blob/main/dagger-core/src/main/java/io/odpf/dagger/core/processors/external/http/HttpSourceConfig.java#L22), it's also a string

> How will this look like ? Can you give an example value ?

I've added test case [here](https://github.com/odpf/dagger/pull/217/commits/1334756642f2a3246070b0c8c90aa74c70825f5c#diff-c7a2518814bf447e7ab2b82bce29b92285113176b5acbb3163d8e7ea3a9d7c38R295) on how will it looks like


> This may cause a NullPointerException

true, updated to (!StringUtil.isNullOrEmpty(httpSourceConfig.getEndpointVariables()))

Reply:
> Can you point me where requestVariables is an Object[]? Since based on [this](https://github.com/odpf/dagger/blob/main/dagger-core/src/main/java/io/odpf/dagger/core/processors/external/http/HttpSourceConfig.java#L22), it's also a string

My bad. I meant `dynamicHeaderVariablesValues` and `requestVariablesValues`. 

If you check [here](https://github.com/odpf/dagger/blob/main/dagger-core/src/main/java/io/odpf/dagger/core/processors/external/http/HttpAsyncConnector.java#L97), we parse the string input such as `ExternalPostProcessorVariableType.HEADER_VARIABLES` and `ExternalPostProcessorVariableType.REQUEST_VARIABLES` into an Object[].  This is because user can specify multiple variables in the string like `customer_name, order_number`.

It depends on whether we should allow user to configure multiple variables in the endpoint also. I think it should be allowed :-) 


> true, updated to (!StringUtil.isNullOrEmpty(httpSourceConfig.getEndpointVariables()))
If we make it as an Object[ ], this check won't be needed anymore.

Reply:
updated the PR @Meghajit

Reply:
Can you add a unit test for endpoint with multiple variables as well ?

Reply:
https://github.com/odpf/dagger/pull/217/files#diff-068c730b8c7790e141bb473ed08ae9ef6d60558d51288be41c10a5dcc9754321R67 added one here

Reply:
Thanks. Looks fine.